corehq/apps/api/resources/__init__.py

```diff
@@ -73,7 +73,7 @@ def method_check(self, request, allowed=None):
             allowed = []
 
         request_method = request.method.lower()
-        allows = ','.join(map(lambda x: x.upper() if x else '', allowed))
+        allows = ','.join([x.upper() if x else '' for x in allowed])
```

Comment:
Approving, because this is correct in that it preserves existing behaviour, but the original code doesn't do what it should. `allows` [should end up](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Allow) with a value like `"GET,POST"` but this code could result in a value like `",,,GET,,POST"`. It should rather be
```python
allows = ','.join([x.upper() for x in allowed if x])
```

Reply:
Also, you can drop the square brackets and just pass a generator expression to `join`